### ISSUE TITLE
Scheduler and Healthcheck stuck Initializing v1.2.2

### ISSUE THREAD

**Issue by goffinf** [rancher/rancher#7192]
**Rancher Version:**

1.2.2

**Docker Version:**

1.12.3 (also tried with 1.10.3 - same behaviour)

**OS and where are the hosts located? (cloud, bare metal, etc):**

RHEL 7.3
AWS

**Setup Details: (single node rancher vs. HA rancher, internal DB vs. external DB)**

Single Rancher sever + 2 hosts (simple set-up to test this problem)

**Environment Type: (Cattle/Kubernetes/Swarm/Mesos)**

Cattle

**Steps to Reproduce:**

1 * Rancher SERVER
2 * Rancher HOST

Spin up instances from AMI

SSH onto Rancher HA SERVER :-

mkdir -p /home/ec2-user/rancher/mysql
chmod -R 0755 /home/ec2-user/rancher

sudo docker run -d -v /home/ec2-user/rancher/mysql:/var/lib/mysql --restart=unless-stopped -p 8080:8080 rancher/server

Browse to the IP:8080 of the HA server

REGISTER HOSTS (Custom)

Spin up a simple stack (nginx) :
Create a global load balancer register nginx service

**Results:** 

Health-check and scheduler containers stuck in 'Initializing' state.
Global LB stuck in 'Initializing' state

**Expected:**

Health-check and scheduler containers should start normally.
Glocal LB should start normally

Clearly a number of users are experiencing this issue. Here are the issues that I've seen recently registered (and have commented on some) 

https://github.com/rancher/rancher/issues/7132
https://github.com/rancher/rancher/issues/7131
https://github.com/rancher/rancher/issues/7130
https://github.com/rancher/rancher/issues/7156
https://github.com/rancher/rancher/issues/7171
https://github.com/rancher/rancher/issues/7190

I understand that this was noted by Rancher and some networking fixes have been introduced in v1.2.2 (I can see the network-manager version has gone up to 0.2.18 and dns to v0.11.0) but the behaviour is the same as it was with Rancher v1.2.1.

I can ping between the ip-sec containers sucessfully, but NOT between ip-sec and any other container, not any other pair of containers.

ICMP **is** enabled on the hosts and server

Have tried with firewalld enabled and disabled (default)

This is an absolute blocker for us.

**Comment by ibuildthecloud** [comment]
@goffinf Traffic flows from container on host A -> ipsec host A -> ipsec host B -> container on host B.  If you can execute a shell on ipsec A and ping ipsec B then ipsec is working correctly and we have established the network.  From what you have described traffic does not flow from ipsec B -> container B.  Furthermore you are saying that any two containers on a single host can not talk to each other?  If that is the case lets just focus on that, because that will be the root of your issue.

So just to clarify, if you run two containers on a single host and they get 10.42.0.10 and 10.42.0.11 for example, If you create a shell on 10.42.0.10, does ping to 10.42.0.11 not work?

Regardless of the answer of that can you please share `iptables-save` output so that I can see what iptables rules you have.  Also the output of `docker version`, `docker info` and the command line arguments to `docker` and the `/etc/docker/daemon.json` if you have one.

**Comment by goffinf** [comment]
Yes your understanding of the network behaviours at present is exactly right.

The info you requested (and anything else that could be useful) 

```
[ec2-user@ip-10-63-25-120 ~]$ docker --version
Docker version 1.12.3, build 6b644ec
```
This is from a brand new EC2 docker instance BEFORE I have deployed Rancher server (the same AMI is used for hosts also)

```
[ec2-user@ip-10-63-25-120 ~]$ docker info
Containers: 0
 Running: 0
 Paused: 0
 Stopped: 0
Images: 1
Server Version: 1.12.3
Storage Driver: devicemapper
 Pool Name: docker-thinpool
 Pool Blocksize: 524.3 kB
 Base Device Size: 10.74 GB
 Backing Filesystem: xfs
 Data file:
 Metadata file:
 Data Space Used: 1.304 GB
 Data Space Total: 102 GB
 Data Space Available: 100.7 GB
 Metadata Space Used: 544.8 kB
 Metadata Space Total: 1.07 GB
 Metadata Space Available: 1.069 GB
 Thin Pool Minimum Free Space: 10.2 GB
 Udev Sync Supported: true
 Deferred Removal Enabled: true
 Deferred Deletion Enabled: false
 Deferred Deleted Device Count: 0
 Library Version: 1.02.135-RHEL7 (2016-09-28)
Logging Driver: json-file
Cgroup Driver: cgroupfs
Plugins:
 Volume: local
 Network: host null bridge overlay
Swarm: inactive
Runtimes: runc
Default Runtime: runc
Security Options: seccomp
Kernel Version: 3.10.0-514.2.2.el7.x86_64
Operating System: Red Hat Enterprise Linux Server 7.3 (Maipo)
OSType: linux
Architecture: x86_64
CPUs: 2
Total Memory: 3.451 GiB
Name: ip-10-63-25-120.eu-west-1.compute.internal
ID: MH3Q:EI23:TOO5:3CVE:QTTX:LODF:QXNT:JOCA:SD2Y:6SIU:HLUI:LQZZ
Docker Root Dir: /var/lib/docker
Debug Mode (client): false
Debug Mode (server): false
Http Proxy: http://management-proxy.nonprod.management.aws-euw1-services.com:80
Https Proxy: http://management-proxy.nonprod.management.aws-euw1-services.com:80
No Proxy: localhost,127.0.0.1,172.17.*,10.*,.avivacloud.com,.aws-euw1-services.com,.via.novonet,.avivagroup.com,.avivaaws.com,.ecs.com,.compute.internal,169.254.169.254
Registry: https://index.docker.io/v1/
Insecure Registries:
 127.0.0.0/8
```

```
[ec2-user@ip-10-63-25-120 ~]$ sudo iptables-save
# Generated by iptables-save v1.4.21 on Sat Dec 24 08:36:05 2016
*nat
:PREROUTING ACCEPT [1:60]
:INPUT ACCEPT [1:60]
:OUTPUT ACCEPT [157:11486]
:POSTROUTING ACCEPT [157:11486]
:DOCKER - [0:0]
-A PREROUTING -m addrtype --dst-type LOCAL -j DOCKER
-A OUTPUT ! -d 127.0.0.0/8 -m addrtype --dst-type LOCAL -j DOCKER
-A POSTROUTING -s 172.17.0.0/16 ! -o docker0 -j MASQUERADE
-A DOCKER -i docker0 -j RETURN
COMMIT
# Completed on Sat Dec 24 08:36:05 2016
# Generated by iptables-save v1.4.21 on Sat Dec 24 08:36:05 2016
*filter
:INPUT ACCEPT [1656:1758209]
:FORWARD ACCEPT [0:0]
:OUTPUT ACCEPT [1612:1535969]
:DOCKER - [0:0]
:DOCKER-ISOLATION - [0:0]
-A FORWARD -j DOCKER-ISOLATION
-A FORWARD -o docker0 -j DOCKER
-A FORWARD -o docker0 -m conntrack --ctstate RELATED,ESTABLISHED -j ACCEPT
-A FORWARD -i docker0 ! -o docker0 -j ACCEPT
-A FORWARD -i docker0 -o docker0 -j ACCEPT
-A DOCKER-ISOLATION -j RETURN
COMMIT
# Completed on Sat Dec 24 08:36:05 2016
```

I don't have a daemon.json at that location, but if you are asking for the config of the docker daemon here are the 2 drop-in files I use and the output of systemctl status docker ..
```

[ec2-user@ip-10-63-25-120 ~]$ systemctl status docker -l
● docker.service - Docker Application Container Engine
   Loaded: loaded (/usr/lib/systemd/system/docker.service; enabled; vendor preset: disabled)
  Drop-In: /etc/systemd/system/docker.service.d
           └─docker.conf, http-proxy.conf
   Active: active (running) since Sat 2016-12-24 08:22:34 GMT; 16min ago
     Docs: https://docs.docker.com
 Main PID: 957 (dockerd)
   Memory: 75.6M
   CGroup: /system.slice/docker.service
           ├─ 957 dockerd --storage-driver=devicemapper --storage-opt dm.thinpooldev=/dev/mapper/docker-thinpool --storage-opt dm.fs=xfs --storage-opt dm.use_deferred_removal=true --log-driver=json-file --log-opt max-size=10m --log-opt max-file=5 -H unix:///var/run/docker.sock
           └─2273 docker-containerd -l unix:///var/run/docker/libcontainerd/docker-containerd.sock --shim docker-containerd-shim --metrics-interval=0 --start-timeout 2m --state-dir /var/run/docker/libcontainerd/containerd --runtime docker-runc

Dec 24 08:22:29 ip-10-63-25-120.eu-west-1.compute.internal docker[957]: time="2016-12-24T08:22:29.764737262Z" level=info msg="libcontainerd: new containerd process, pid: 2273"
Dec 24 08:22:32 ip-10-63-25-120.eu-west-1.compute.internal docker[957]: time="2016-12-24T08:22:32.136533838Z" level=info msg="Graph migration to content-addressability took 0.00 seconds"
Dec 24 08:22:32 ip-10-63-25-120.eu-west-1.compute.internal docker[957]: time="2016-12-24T08:22:32.137920640Z" level=info msg="Loading containers: start."
Dec 24 08:22:33 ip-10-63-25-120.eu-west-1.compute.internal docker[957]: time="2016-12-24T08:22:33.566837043Z" level=info msg="Firewalld running: false"
Dec 24 08:22:33 ip-10-63-25-120.eu-west-1.compute.internal docker[957]: time="2016-12-24T08:22:33.988500646Z" level=info msg="Default bridge (docker0) is assigned with an IP address 172.17.0.0/16. Daemon option --bip can be used to set a preferred IP address"
Dec 24 08:22:34 ip-10-63-25-120.eu-west-1.compute.internal docker[957]: time="2016-12-24T08:22:34.041592483Z" level=info msg="Loading containers: done."
Dec 24 08:22:34 ip-10-63-25-120.eu-west-1.compute.internal docker[957]: time="2016-12-24T08:22:34.041710903Z" level=info msg="Daemon has completed initialization"
Dec 24 08:22:34 ip-10-63-25-120.eu-west-1.compute.internal docker[957]: time="2016-12-24T08:22:34.041726894Z" level=info msg="Docker daemon" commit=6b644ec graphdriver=devicemapper version=1.12.3
Dec 24 08:22:34 ip-10-63-25-120.eu-west-1.compute.internal docker[957]: time="2016-12-24T08:22:34.050268583Z" level=info msg="API listen on /var/run/docker.sock"
Dec 24 08:22:34 ip-10-63-25-120.eu-west-1.compute.internal systemd[1]: Started Docker Application Container Engine.
```
```

[ec2-user@ip-10-63-25-120 ~]$ sudo cat /etc/systemd/system/docker.service.d/docker.conf
[Service]
ExecStart=
ExecStart=/usr/bin/docker daemon --storage-driver=devicemapper --storage-opt dm.thinpooldev=/dev/mapper/docker-thinpool --storage-opt dm.fs=xfs --storage-opt dm.use_deferred_removal=true --log-driver=json-file --log-opt max-size=10m --log-opt max-file=5 -H unix:///var/run/docker.sock $OPTIONS
```
```

[ec2-user@ip-10-63-25-120 ~]$ sudo cat /etc/systemd/system/docker.service.d/http-proxy.conf
[Service]
Environment="HTTP_PROXY=http://management-proxy.nonprod.management.aws-euw1-services.com:80"
Environment="http_proxy=http://management-proxy.nonprod.management.aws-euw1-services.com:80"
Environment="HTTPS_PROXY=http://management-proxy.nonprod.management.aws-euw1-services.com:80"
Environment="https_proxy=http://management-proxy.nonprod.management.aws-euw1-services.com:80"
Environment="NO_PROXY=localhost,127.0.0.1,172.17.*,10.*,.avivacloud.com,.aws-euw1-services.com,.via.novonet,.avivagroup.com,.avivaaws.com,.ecs.com,.compute.internal,169.254.169.254"
Environment="no_proxy=localhost,127.0.0.1,172.17.*,10.*,.avivacloud.com,.aws-euw1-services.com,.via.novonet,.avivagroup.com,.avivaaws.com,.ecs.com,.compute.internal,169.254.169.254"
```

Other info ...

1.  This is a simple test set-up. 1 Rancher server with a host bind mounted MySQL Db volume + 2 hosts. So no ALB for the Rancher servers (I'm just running a single server and a couple of hosts to test out this problem).
2.  Ports on the Rancher SERVER and HOSTs are all the same (I would normally use different specific security groups but since this is just a test I'm using the same build for both). So all instances have the following INGRESS ports open 

```
Port        Protocol        Source 
----        ---------        ------
-1          icmp               10.0.0.0/8
22          tcp                 10.0.0.0/8
80          tcp                 10.0.0.0/8
500        udp                10.0.0.0/8
4118      tcp                 10.0.0.0/8 (internal mgmt)
4500      udp                10.0.0.0/8
8080      tcp                 10.0.0.0/8
9345      tcp                 10.0.0.0/8
```

2.  The firewalld service is DISABLED
3.  We have a policy of only sourcing docker images from an internal (ECR) registry. The following Rancher images versions are used ...

```
rancher/server:v1.2.2

rancher/agent:v1.1.2
rancher/network-manager:v0.2.19
rancher/dns:v0.11.0
rancher/lb-service-haproxy:v0.4.6
rancher/net:v0.7.5
rancher/metadata:v0.6.8
rancher/healthcheck:v0.1.0
```

Regards

Fraser.

**Comment by goffinf** [comment]
Eh ? ... something strange happening ....

When I registered a host to the Rancher sever, the agent came up as expected but NONE of the other infrastructure stack containers launched ?

Then when I attempted to create a stack for nginx, this error ...

`Activating (Failed to allocate instance [container:1i13]: Bad instance [container:1i13] in state [removed]: Failed to find network for networkMode managed) `

**Comment by goffinf** [comment]
WTF .... now the infrastructure stack has come up and they are all started .. but 13 MINUTES after the rancher/agent ... so did the previously failed nginx stack

```
[ec2-user@ip-10-63-25-214 ~]$ docker ps -a
CONTAINER ID        IMAGE                                                     COMMAND                  CREATED             STATUS                      PORTS               NAMES
56d777917ba2        rancher/lb-service-haproxy:v0.4.6                         "/.r/r /tini -- lb-co"   2 minutes ago       Up 2 minutes                                    r-global-lb-global-lb-1-90b93d02
805ff847fbd0        906261169288.dkr.ecr.eu-west-1.amazonaws.com/nginx:1.11   "/.r/r nginx -g 'daem"   9 minutes ago       Up 9 minutes                                    r-nginx-ecr-nginx-ecr-1-05745e5a
b449d5dc50d3        rancher/net:v0.7.5                                        "/.r/r start.sh"         10 minutes ago      Up 10 minutes                                   r-ipsec-ipsec-1-e5e47c1f
447233a99574        rancher/dns:v0.11.0                                       "rancher-dns --metada"   10 minutes ago      Up 10 minutes                                   r-network-services-metadata-dns-1-c4c77291
5fa69b1cbf85        rancher/net:v0.7.5                                        "sh -c 'touch /var/lo"   10 minutes ago      Up 10 minutes                                   r-ipsec-ipsec-cni-driver-1-766a55f7
310844edc11e        rancher/metadata:v0.6.8                                   "start.sh rancher-met"   10 minutes ago      Up 10 minutes                                   r-network-services-metadata-1-175c2a2c
570bba183e4a        rancher/scheduler:v0.4.0                                  "/.r/r scheduler"        10 minutes ago      Up 10 minutes                                   r-scheduler-scheduler-1-5556795a
d59eb6e01c9a        rancher/healthcheck:v0.1.0                                "/.r/r /tini -- healt"   10 minutes ago      Up 10 minutes                                   r-healthcheck-healthcheck-1-44cf4bd8
da35c28365b1        rancher/network-manager:v0.2.19                           "/usr/bin/start.sh pl"   10 minutes ago      Up 10 minutes                                   r-network-services-network-manager-1-c65d6839
495d33903582        rancher/agent:v1.1.2                                      "/run.sh run"            23 minutes ago      Up 23 minutes                                   rancher-agent
1657abe5511f        rancher/agent:v1.1.2                                      "/run.sh http://10.63"   23 minutes ago      Exited (0) 23 minutes ago                       infallible_poitras
```

Then I added a new host, and the infrastructure stack there launched immediately.

I created a global load balancer service and pointed to nginx and that came up clean too and routes as you would expect from either host to the nginx container.

Everything is now working BUT .... why the delay in the infrastructure stack launch ... and why is everything now working as it should (I hate it when things are unexplained and it seems likely to me this is going to happen again, so I need to understand whats going on here).

Ok, since you are going to ask .... what did I change .... well ... I did a few things but I'm not convinced any of them were causal but here goes ...

I tried a curl to the Rancher server from the first host (I only had one started at that point) ....

curl http://10.63.25.120

 and it failed with a Squid proxy error ... however I do have 10.* in both the host level NO_PROXY and for the docker daemon. But, just for kicks, I added it in both places and sure enough the curl command then worked.

Then I did a pull of an image from ECR from an SSH session on the host to check that was working OK. It was and the pull was successful.

I deactivated and then removed the host from Rancher server.

Then I removed all containers using ...

`docker rm -f -v $(docker ps -a -q)`

The attempted to remove /var/lib/rancher with ...

```
sudo systemctl docker stop
sudo rm -rf /var/lib/rancher
```

That failed with a 'Device busy ...' message so I did a sudo reboot and tried again .. success

Re-registered the host

Messed about a bit with inoccuous stuff for a while then ... boom I noticed that the stacks had all launched !!!!????

SSH'd to my other hosts and ran the 'Add Custom Host' docker command .. I did NOT change anything to do with NO_PROXY on this host (it does NOT have the IP of the Rancher server either in the host NO_PROXY or the docker daemon config) ... the host launched and the infrastructure stack came up right away ... healthcheck running on both hosts.

I can successful ping between ANY pair of containers on the same or different hosts

Iptables on one of the HOSTs (the Rancher server is unchanged) looks like this ..

```
[ec2-user@ip-10-63-25-214 ~]$ sudo iptables-save
# Generated by iptables-save v1.4.21 on Sat Dec 24 10:30:12 2016
*nat
:PREROUTING ACCEPT [21:1608]
:INPUT ACCEPT [2:120]
:OUTPUT ACCEPT [120:8803]
:POSTROUTING ACCEPT [135:10039]
:CATTLE_HOSTPORTS_POSTROUTING - [0:0]
:CATTLE_NAT_POSTROUTING - [0:0]
:CATTLE_OUTPUT - [0:0]
:CATTLE_POSTROUTING - [0:0]
:CATTLE_PREROUTING - [0:0]
:DOCKER - [0:0]
-A PREROUTING -m addrtype --dst-type LOCAL -j CATTLE_PREROUTING
-A PREROUTING -m addrtype --dst-type LOCAL -j DOCKER
-A OUTPUT -m addrtype --dst-type LOCAL -j CATTLE_OUTPUT
-A OUTPUT ! -d 127.0.0.0/8 -m addrtype --dst-type LOCAL -j DOCKER
-A POSTROUTING -j CATTLE_HOSTPORTS_POSTROUTING
-A POSTROUTING -j CATTLE_NAT_POSTROUTING
-A POSTROUTING -s 172.17.0.0/16 ! -o docker0 -j MASQUERADE
-A CATTLE_HOSTPORTS_POSTROUTING -s 10.42.65.152/32 -d 10.42.65.152/32 -p udp -m udp --dport 500 -j MASQUERADE
-A CATTLE_HOSTPORTS_POSTROUTING -s 10.42.65.152/32 -d 10.42.65.152/32 -p udp -m udp --dport 4500 -j MASQUERADE
-A CATTLE_HOSTPORTS_POSTROUTING -s 10.42.103.214/32 -d 10.42.103.214/32 -p tcp -m tcp --dport 80 -j MASQUERADE
-A CATTLE_NAT_POSTROUTING -s 10.42.0.0/16 ! -o docker0 -p tcp -j MASQUERADE --to-ports 1024-65535
-A CATTLE_NAT_POSTROUTING -s 10.42.0.0/16 ! -o docker0 -p udp -j MASQUERADE --to-ports 1024-65535
-A CATTLE_NAT_POSTROUTING -s 10.42.0.0/16 ! -o docker0 -j MASQUERADE
-A CATTLE_NAT_POSTROUTING -o docker0 -m addrtype --src-type LOCAL --dst-type UNICAST -j MASQUERADE
-A CATTLE_OUTPUT -p udp -m udp --dport 500 -m addrtype --dst-type LOCAL -j DNAT --to-destination 10.42.65.152:500
-A CATTLE_OUTPUT -p udp -m udp --dport 4500 -m addrtype --dst-type LOCAL -j DNAT --to-destination 10.42.65.152:4500
-A CATTLE_OUTPUT -p tcp -m tcp --dport 80 -m addrtype --dst-type LOCAL -j DNAT --to-destination 10.42.103.214:80
-A CATTLE_PREROUTING ! -i docker0 -p udp -m udp --dport 500 -j MARK --set-xmark 0x1068/0xffffffff
-A CATTLE_PREROUTING ! -i docker0 -p udp -m udp --dport 500 -j DNAT --to-destination 10.42.65.152:500
-A CATTLE_PREROUTING -p udp -m udp --dport 500 -m addrtype --dst-type LOCAL -j DNAT --to-destination 10.42.65.152:500
-A CATTLE_PREROUTING ! -i docker0 -p udp -m udp --dport 4500 -j MARK --set-xmark 0x1068/0xffffffff
-A CATTLE_PREROUTING ! -i docker0 -p udp -m udp --dport 4500 -j DNAT --to-destination 10.42.65.152:4500
-A CATTLE_PREROUTING -p udp -m udp --dport 4500 -m addrtype --dst-type LOCAL -j DNAT --to-destination 10.42.65.152:4500
-A CATTLE_PREROUTING ! -i docker0 -p tcp -m tcp --dport 80 -j MARK --set-xmark 0x1068/0xffffffff
-A CATTLE_PREROUTING ! -i docker0 -p tcp -m tcp --dport 80 -j DNAT --to-destination 10.42.103.214:80
-A CATTLE_PREROUTING -p tcp -m tcp --dport 80 -m addrtype --dst-type LOCAL -j DNAT --to-destination 10.42.103.214:80
-A DOCKER -i docker0 -j RETURN
COMMIT
# Completed on Sat Dec 24 10:30:12 2016
# Generated by iptables-save v1.4.21 on Sat Dec 24 10:30:12 2016
*filter
:INPUT ACCEPT [2091:661109]
:FORWARD ACCEPT [0:0]
:OUTPUT ACCEPT [2287:1263761]
:CATTLE_FORWARD - [0:0]
:DOCKER - [0:0]
:DOCKER-ISOLATION - [0:0]
-A FORWARD -j CATTLE_FORWARD
-A FORWARD -j DOCKER-ISOLATION
-A FORWARD -o docker0 -j DOCKER
-A FORWARD -o docker0 -m conntrack --ctstate RELATED,ESTABLISHED -j ACCEPT
-A FORWARD -i docker0 ! -o docker0 -j ACCEPT
-A FORWARD -i docker0 -o docker0 -j ACCEPT
-A CATTLE_FORWARD -m mark --mark 0x1068 -j ACCEPT
-A DOCKER-ISOLATION -j RETURN
COMMIT
# Completed on Sat Dec 24 10:30:12 2016
```

Can you explain this ?

**Comment by ibuildthecloud** [comment]
If you don't have any sensitive information in this environment, could you send me the DB dump (`mysqldump`)?  darren@rancher.com  Then I can tell you exactly what it did.  Where are you configuring the proxy?  For rancher we mostly care about proxy configuration being set in 2 places.  As docker run env variables on rancher/server and env variable on the docker run command for rancher/agent to add a custom host.  Does a proxy exist in between the Rancher server and the hosts?

**Comment by goffinf** [comment]
No sensitive data, this is just a test set-up to try and sort out this particular issue.

Not sure about the MySQL dump though. I kept things simple here and didn't create a HA environment or an external Db, just used a single Rancher server with the Db bind mounted to the Rancher server, per the docs ...

`sudo docker run -d -v /home/ec2-user/rancher/mysql:/var/lib/mysql --restart=unless-stopped -p 8080:8080 rancher/server`

 I'll zip up that whole directory structure and mail it to you in case that helps ?

Proxy.... Again in this (simple) case I just started Rancher itself as above and then added custom hosts from the available command ...

`sudo docker run -d --privileged -v /var/run/docker.sock:/var/run/docker.sock -v /var/lib/rancher:/var/lib/rancher rancher/agent:v1.1.2 http://10.63.25.120:8080/v1/scripts/0DC83598D7FB66F2A053:1482670800000:MimxprUQSSrzMDtfBaR0Ur0d0F4`

I didn't add any environment variables or any LB/proxy between the Rancher server and hosts to start with. Once the hosts registered successfully (which was where I was seeing the stuck 'initializing' state) I created a Rancher global scope LB/Proxy to run on all hosts and registered my Nginx service (again, until this last attempt, the LB was also getting stuck 'Initializing').

In the normal config, we have Rancher running in a 3 node HA cluster with an ALB above. The Rancher servers themsleves run in an ASG.  Hosts run under a classis ELB and are just registered as custom and we **were** also using some external proxy instances, but TBH we intend to dispense with those this time around and just run a Rancher global LB to provide the reverse proxy to services (we only want to expose the Proxy/LB port(s) publically and no others.

Is the lack of those env variables critical (I don't recall reading about that in the docs ?) ?

HTHs

Fraser.

**Comment by goffinf** [comment]
Looking at the Dockerfile and run.sh for the rancher/agent it might be that I misunderstood your comment about 'proxy' ... were you talking about the HTTP_PROXY, HTTPS_PROXY and perhaps more importantly the NO_PROXY environment variables. If so, then I can tell you that I set these (in upper and lower case) both on the docker daemon (using the drop-in file http-proxy.conf mentioned above) and as environment variables on the host. Interestingly when I built the AMI I took out a script that I used previously that set these in .profile, wget and yum.conf, ... perhaps that was a mistake ?

Fraser.

**Comment by goffinf** [comment]
Spun up a new environment but same result I'm afraid (whats that definition of madness again ;-)  )

I have this for Rancher Server ..

ALB -> Listener (443) -> Target Group (8080) -> ASG/LC (min/max/des=1) -> Rancher Server *1

and for Hosts ...

ALB  -> Listener (80) -> Target Group (80) - ASG/LC (min/max/des=2) -> Host * 2

Target Group health-checks pass and both host instances are healthy.

Can successfully reach the Rancher UI (ALB cert installed in browser)

Successfuly registered hosts the the https Rancher Server endpoint

Host 1: Scheduler container state 'Initialzing' - Healthcheck 'Started'
Host 2: Healthcheck container status 'Initializing'

All other infrastructure service containers started.

Can ping between IIPSec containers
Can't ping from IpSec container to any other container (except IpSec)
Can't ping from any non IpSec container to any other non IpSec container.

Clearly theres a network issue here but I've run out of ideas

**Comment by JaviSabalete** [comment]
+1

**Comment by goffinf** [comment]
This issue has now been resolved for us at least with some great assistance from Rancher support in the US and U.K. After replicating our exact environment (using rpm -qa) with the exception of 2 packages and demonstrating that Rancher worked as expected the problem was narrowed down to the Trend Deep Security Agent that we run as standard on our hardened AMIs. After temporarily disabling the agent Rancher was able to successful communicate between containers on the same host and across hosts. The scheduler and health-check containers from the infrastructure stack started and none was left in initialising state. So the cause of this was that packets were being blocked by DSA. Fortunately there is a documented configuration for DSA that allows it to work with docker which we will now be implementing (see: https://help.deepsecurity.trendmicro.com/using-deep-security-with-docker.html).

I appreciate that this solution may not be relevant to other Rancher users but might at least help point the way to the potential cause (something blocking container traffic).

HTHs

Fraser.
